DECLARING_DEPENDENCIES.md

```diff
@@ -0,0 +1,210 @@
+- [google-cloud-java](https://github.com/googleapis/google-cloud-java)
+- [beam](https://github.com/apache/beam)
+
+## Ensuring Compatibility
+
+If you depend on one or more of these libraries, in order to
+ensure that your own project uses compatible versions of them, follow
+the guidance below for your build system.
+
+### Maven
+
+#### Use the requireUpperBoundDeps enforcer rule
+
+Dependency conflicts can happen with Maven even when libraries follow
+semver because Maven's dependency mediation algorithm can select older
+versions instead of newer versions, meaning new features will be
+missing that other libraries depend on. The `requireUpperBoundDeps`
+enforcer rule can be used to automatically discover incorrect version
+selection because it fails the build if anything other than the most
+recent version is chosen.
+
+You can add `requireUpperBoundDeps` to your build like this:
+
+```
+  <build>
+    <plugins>
+      <plugin>
+        <groupId>org.apache.maven.plugins</groupId>
+        <artifactId>maven-enforcer-plugin</artifactId>
+        <version>3.0.0-M2</version>
+        <executions>
+          <execution>
+            <id>enforce</id>
+            <goals>
+              <goal>enforce</goal>
+            </goals>
+            <configuration>
+              <rules>
+                <requireUpperBoundDeps/>
+              </rules>
+            </configuration>
+          </execution>
+        </executions>
+      </plugin>
+    </plugins>
+  </build>
+```
+
+When the newest versions are not compatible, you can use BOMs to
+select compatible versions.
+
+#### Import BOMs to ensure dependencies are consistent
+
+Conflicts can occur when multiple artifacts from a single library are
+part of a dependency tree, and Maven resolves different artifacts from
+that library to versions from different releases. Using a BOM fixes
+this problem because a BOM dictates consistent versions for all
+artifacts from a library.
+
+You can use a BOM like this - this example is for `google-cloud-bom`:
+
+```
+  <dependencyManagement>
+    <dependencies>
+      <dependency>
+        <groupId>com.google.cloud</groupId>
+        <artifactId>google-cloud-bom</artifactId>
+        <version>0.81.0-alpha</version>
+        <type>pom</type>
+        <scope>import</scope>
+       </dependency>
+     </dependencies>
+  </dependencyManagement>
+
+  <dependencies>
+    <dependency>
+      <groupId>com.google.cloud</groupId>
+      <artifactId>google-cloud-storage</artifactId>
+    </dependency>
+  </dependencies>
+```
+
+In the example above, since the BOM manages library versions, the
+version of the google-cloud-storage artifact is omitted.
+
+See the "Choosing BOMs" section below if you need to import more than one BOM.
+
+### Gradle
+
+Gradle automatically selects the highest version in the dependency
+tree of any given dependency.
+
+#### Import BOMs to ensure your dependencies are consistent
+
+It is possible to accidentally use versions of different artifacts
```

Comment:
"It is possible to accidentally use versions of different artifacts
from a library from different releases and experience a dependency
conflict."

This is unclear. Maybe, "When a library publishes multiple artifacts, using different  versions for different artifacts may cause dependency
conflicts."